src/Microsoft.Data.SqlClient/tests/UnitTests/IO/TdsWriteStreamTest.cs

```diff
@@ -0,0 +1,292 @@
+                    Assert.Equal(lengthRead, packetSize);
+                }
+
+                // Match content 
+                int contentLength = lengthRead - 8;
+                byte[] expectedData = inputBuffer.AsSpan().Slice(matchOffset, contentLength).ToArray();
+                Assert.Equal(expectedData, header.Content);
+                // Move the match offset to the next packet.
+                matchOffset += contentLength;
+            } while (header?.Status != TdsEnums.ST_EOM);
+        }
+
+        [Fact]
+        public async void WriteStream_WriteMultiplePacketsAsync()
+        {
+            int packetSize = 20;
+            using MemoryStream ms = new();
+            using TdsWriteStream tdsWriteStream = new(ms);
+            tdsWriteStream.SetPacketSize(packetSize);
+
+            tdsWriteStream.PacketHeaderType = TdsStreamPacketType.Login7;
+            byte[] inputBuffer = new byte[packetSize*2];
+            for (int i = 0; i < inputBuffer.Length; i++)
+            {
+                inputBuffer[i] = (byte)i;
+            }
+
+            // Multiple packets will be created and flushed.
+            await tdsWriteStream.WriteAsync(inputBuffer.AsMemory(), CancellationToken.None);
+            await tdsWriteStream.FlushAsync();
+
+            ms.Position = 0;
+            int matchOffset = 0;
+
+            // Allocate twice the needed buffer.
+            byte[] memoryStreamReadBuffer = new byte[inputBuffer.Length * 2];
+            TdsPacketHeader? header = null;
+            do
+            {
+                int lengthRead = ms.Read(memoryStreamReadBuffer, 0, packetSize);
+                Assert.True(lengthRead >= TdsEnums.HEADER_LEN, "The total length read was less than the header of TDS packet.");
+                header = new(memoryStreamReadBuffer);
+
+                Assert.Equal((byte)TdsStreamPacketType.Login7, header.PacketType);
+
+                // Assert the packet type is either ST_EOM or ST_BATCH.
+                Assert.Contains(header.Status, _packetStatuses);
+                if (header.Status == TdsEnums.ST_BATCH)
+                {
+                    Assert.Equal(packetSize, header.Length);
+                    // When Batch packets are sent out, they are expected to be completely full to the packet size.
+                    Assert.Equal(lengthRead, packetSize);
+                }
+                // Match content 
+                int contentLength = lengthRead - 8;
+                byte[] expectedData = inputBuffer.AsSpan().Slice(matchOffset, contentLength).ToArray();
+                Assert.Equal(expectedData, header.Content);
+                // Move the match offset to the next packet.
+                matchOffset += contentLength;
+            } while(header?.Status != TdsEnums.ST_EOM);
+        }
+
+        [Fact]
+        public async void WriteStream_WriteSinglePacketAsync()
```

Comment:
Nit: Please add inline comments where possible, so intent of test is readable..

Reply:
Will do. Thanks.